### ISSUE TITLE
Libre-SOC

### ISSUE THREAD

**Issue by fricklerhandwerk** [ngi-nix/ngipkgs#11]
A libre processor core design.

- [NLnet project](https://nlnet.nl/project/Libre-SOC-HPC)
- [source code](https://git.libre-soc.org/)
- [manual build setup](https://git.libre-soc.org/?p=dev-env-setup.git;a=blob;f=README.mdwn;h=5265d50534f86fe7b930b9582fe0f156d076ab7a;hb=HEAD)
- [Nix build setup](https://git.libre-soc.org/?p=soc.git;a=tree;f=nix;h=0a9bf679252c711cfbeae886781442bb5cfd347f;hb=8e00e09b914b9fa3af430c2f913a9921207063cd)
- related work:
  - https://github.com/ngi-nix/ngi/issues/234
  - https://github.com/ngi-nix/ngi/issues/145
  - https://github.com/ngi-nix/libresoc-soc
  - https://github.com/ngi-nix/coriolis/blob/devel/flake.nix
  
This project is huge. Some dependencies are already in Nixpkgs, some are upstream, some are in [`ngi-nix`](https://github.com/ngi-nix) repositories, some apparently still unpackaged. This needs to be sorted out and broken down into separate tasks.

The end goal is to build the latest design with one command. It's more likely though that getting components packaged will simply benefit other hardware designers.

The lead developer seems to be quite frustrated with the Nix community's development model and past interactions with Nix maintainers. I propose that we stay out of the way and just make everything work on our side.

**Comment by fricklerhandwerk** [comment]
@jleightcap thanks a lot for the update! I also think that taking all the derivations and maintaining them here is the most sensible option. We can then ask upstream to remove their Nix code altogether.

**Comment by albertchae** [comment]
> 2b. Include monorepo in this repo
> 
> CC @albertchae, who had this suggestion to effectively fork the upstream monorepo into this repository.

@jleightcap To clarify I didn't mean to fork the entire repo but just the nix derivations while preserving their commit history from the monorepo. This should be possible with a combination of `git filter-branch` https://amandawalkerbrubaker.medium.com/splitting-your-git-repo-while-maintaining-commit-history-35b9f4597514 and the merging with unrelated histores https://github.com/ngi-nix/ngipkgs/issues/12#issuecomment-1733510381

But this could be unnecessary work when copying files is a lot easier

**Comment by fricklerhandwerk** [comment]
If it's not too much overhead, I'd prefer preserving history to give credit where it's due.

**Comment by lorenzleutgeb** [comment]
@fricklerhandwerk

> If it's not too much overhead, I'd prefer preserving history to give credit where it's due.

Then don't squash merge next time please.

**Comment by lorenzleutgeb** [comment]
@jleightcap please confirm that this means we can now archive https://github.com/ngi-nix/libresoc-soc

**Comment by fricklerhandwerk** [comment]
@lorenzleutgeb 

> Then don't squash merge next time please. As far as I understand @jleightcap and @albertchae spent quite some energy in preserving a good history in #168. [7f8b64f](https://github.com/ngi-nix/ngipkgs/commit/7f8b64f30137298397202dd7ea5e0dfa0a19f6ba) was a root (obfuscated by a rebase).

@lorenzleutgeb yes, absolutely -- sorry. I looked at the history and was impressed by the second part and confused by the first part, then decided to squash. Re-did it with the exceptional force push to main (forgive my sins).
